src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorageWithProperties/AzureTableStorageEntityFactory.cs

```diff
@@ -61,9 +61,15 @@ public static DynamicTableEntity CreateEntityWithProperties(LogEvent logEvent, I
 
             List<KeyValuePair<ScalarValue, LogEventPropertyValue>> additionalData = null;
             var count = dynamicProperties.Count;
+			bool isNumeric;
+			int parsedNumber;
+
             foreach (var logProperty in logEvent.Properties)
             {
-                if (count++ < _maxNumberOfPropertiesPerRow - 1)
+				isNumeric = int.TryParse(logProperty.Key, out parsedNumber);
```

Comment:
That's a reasonable assumption - though sometimes numeric property names provide a "free" upgrade from text -> structured logs, so other sinks treat them as normal.